Stronger condition to check NFS

Seb35 opened the pull request:
### Description:

Issue: #12217 - in the system check page, filesystems like btrfs subvolumes are wrongly recognised as NFS.

NFS filesystems are identified as 'nfs' or 'nfs4' by `df -T`, so the added condition tests the existence of `' nfs'`.

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

Seb35 commented:
I tested on my Matomo installation where the directory `tmp` (and `tmp/sessions`) are a btrfs subvolume: the check now passes.

diosmosis commented on core/Filesystem.php:
```diff
@@ -161,9 +162,11 @@ public static function checkIfFileSystemIsNFS()
                 if (!$commandFailed
-                    && count($output) > 1) {
+                    && count($outputArray) > 1
+                    && strpos(' nfs', $output) !== false) {
```
Hi @Seb35, instead of looking for a space prefix, can we check for a word boundary using a regex? Just in case the output has tabs.

Seb35 replied:
I changed the condition to `preg_match('/bnfs\d?\b/', $output)` to search "nfs", "nfs4" or any "nfs[0-9]" as a single word.

diosmosis reviewed (changes requested):
Hi @Seb35, thanks for going through the trouble of creating this PR! I've requested a single change, otherwise it looks good to merge!

Seb35 committed:
Stronger condition to check NFS

Exclude unrecognized filesystems like btrfs subvolumes.
NFS filesystems are identified as 'nfs' or 'nfs4'.

Issue: #12217

Seb35 commented:
I did the requested change, it is now more robust.

diosmosis commented:
@Seb35 I tested locally and it works well, thanks again for creating this PR and contributing some code! We certainly appreciate the effort.
